Source/DafnyCore/AST/Types/ArrowType.cs

```diff
@@ -43,6 +41,21 @@ public class ArrowType : UserDefinedType {
     }
   }
 
+  public static void ComputeResolvedOpcodeAndBoundedPool(Expression collection, CollectionType collectionType,
```

Comment:
Should the computation of bounded pools be tied more closely to types? Should we have `CollectionType.GetBoundedPool` or `Type.GetBoundedPool` ? I'm not sure why this code relates mostly to `ArrowType` and I see it's also used outside of `ArrowType`

Reply:
Eagle eyes! :)

I had been a bit unhappy about the placement of this method. Reflecting on it some more, I now think putting it near the various `Expression.Create...` methods, since it produces information that you need when creating an expression of the form `x in C`. So, I moved it there.

Btw, the `boundedPool` out-parameter of this method is not used by any caller in this PR, but it is used in my soon-to-come follow-up PR.